Sources/PointFree/Database.swift

```diff
@@ -69,8 +73,19 @@ public struct Database {
     }
   }
 
+  public struct EpisodeCredit: Decodable {
+    public internal(set) var episodeSequence: Int
+    public internal(set) var userId: User.Id
+
+    public enum CodingKeys: String, CodingKey {
+      case episodeSequence = "episode_sequence"
+      case userId = "user_id"
+    }
+  }
+
   public struct User: Decodable {
     public internal(set) var email: EmailAddress
+    public internal(set) var episodeCreditCount: Int
```

Comment:
One thought is that credits count could be derived from "unfulfilled" credit rows in the credits table...not sure if it's worth the refactor, though.

Reply:
ah so that's not how the episode credits are structured. the "source of truth" of how many credits one has is this field. If this field is >0 you will get the "use credit" button on a subscriber only episode.

The `EpisodeCredit` table holds the credits that were used. So if a row is present for a (episode, user), they get access to that episode.

Does that make sense?

Reply:
@mbrandonw It totally makes sense. I was just suggesting an alternate structure so the single source of truth would be rows in the credits table.